src/00/02-introspect/proposal.adoc

```diff
@@ -0,0 +1,879 @@
+`.stdinfo` text. For example: `cl.exe` => `cl.stdinfo`, or
+`pass:[g++]` => `pass:[g++].stdinfo`.
+
+2. The introspection file can be found: in the same filesystem location
+as the tool executable (or script, or equivalent), in an implementation defined
+location relative to the tool location, or in an implementation defined global
+location (i.e. an absolute path location).
+
+=== Declaration
+
+The consumer can inform, i.e. declare, to the target tool that specific
+capabilities should use particular versions when responding with information
+using one or more `--std-info=<VersionSpec>` options. The declarations can only
+exist in tandem with options for the mentioned capabilities. It's expected that
+a consumer will first _introspect_ a target tool to discover what it supports.
+Followed by the consumer _declaring_ to the target tool what version(s) of the
+capabilities it is willing to consume. The target tool can then either accept
+the declared capability versions or indicate an error.
+
+An exchange between a consumer and target tool would begin with the
+_introspection_:
+
+[source,shell]
+----
+tool "--std-info"
+----
+
+With a target tool response:
+
+[source,json]
+----
+include::example-03.json[]
+----
+
+Which the consumer can use to _declare_ the specific capability versions:
+
+[source,shell]
+----
+tool "--std-decl=std_info=2.0.0" "--std-decl=gcc_extra=2.1.0" ...
+----
+
+=== Levels
+
+For some use cases it helps to simplify the extent of information the
+introspection understands. While it would be reasonable to expect a tool written
+in a modern general purpose programming language to fully implement all aspects
+of the introspection. It would not be practical to have a shell script parse
+and recognize the more challenging aspect of parsing version number ranges and
+matching them together. To support such use cases the introspection has to
+support levels "min" and "full".
+
+Obviously the "full" level equates to the tool understanding all the arguments
+and values. The "min" level only understands these:
+
+* Only introspection `--std-info` option.
+* Single version number in the responses for `--std-info`.
+
+This has the effect that a tool which only support the "min" level can only
+support specific versions of the capabilities it implements. But it also means
+that consumers will need to adjust their behavior to the tool instead of being
+able to ask the tool to adjust to the consumer. Consequently the consumer will
+likely have the more complex logic to do that adjustment.
+
+=== Capabilities
+
+For this proposal capabilities refers to any published coherent target tool
+interface. This can include any single interface, like a single target tool
+option. Or it can include a collective interface of the target tool that covers
+many options. A capability is specified as a series of "scoped" identifiers
+separated by underscore ("_"). The capability must match this regular expression:
```

Comment:
There is plenty of reasoning in this section for the rest of the regex, however there is nothing to explain why `_` is preferred over `-` or `:` or even an ascii C0 control code.